index.js

```diff
@@ -41,7 +50,10 @@ const bufferOutput = (subprocess, result, streamName) => {
 	});
 };
 
-const getOutput = ({stdout, stderr}) => ({
+const getOutput = ({exitCode, signalCode}, {stdout, stderr}) => ({
+	// `exitCode` can be a negative number (`errno`) when the `error` event is emitted on the subprocess
+	...(exitCode === null || exitCode < 0 ? {} : {exitCode}),
+	...(signalCode === null ? {} : {signalName: signalCode}),
```

Comment:
We use `subprocess.exitCode` and `subprocess.signalCode` instead of the values passed to the `close` event emitter. That's because the `close` event is not emitted when an `error` event was emitted before `spawn`.

But it turns out those are the exact same values.
See source: https://github.com/nodejs/node/blob/8b0c699f2aafdf81bc4834b9bd2a4923741d3a6f/lib/internal/child_process.js#L1104